### PR TITLE
Remove `structured_reasons_for_rejection_redesign` feature flag

### PR DESCRIPTION
**Pull request by steventux**
## Context

We've released a redesign of Structured Reasons for Rejection, the feature flag for this is no longer needed.

## Changes proposed in this pull request

- Removes feature flag for redesign of SR4R
- Removes any code conditions which evaluate or depend on the state of the feature flag
- Amends specs to use new form and removes dependencies on feature flag

## Link to Trello card

https://trello.com/c/e9dSB9wB/5048-remove-sr4r-redesign-feature-flag

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Review by cwrw — COMMENTED** [review]
could you add a data migration to also remove the feature flag please?

File: spec/system/provider_interface/reject_an_application_spec.rb [hunk 1]
```diff
@@ -5,34 +5,31 @@
   include ProviderUserPermissionsHelper
   include CourseOptionHelpers
 
-  scenario 'giving reasons for rejection', with_audited: true do
-    FeatureFlag.deactivate(:structured_reasons_for_rejection_redesign)
-
+  scenario 'giving rejection reasons using redesigned rejection form', with_audited: true do
```
Comment (cwrw) [inline spec/system/provider_interface/reject_an_application_spec.rb:8]: for those two system specs just to double check we moved the new R4R from the other specs deleted into the old ones?

Comment (steventux) [reply spec/system/provider_interface/reject_an_application_spec.rb:8]: Yes that's right, `rejection_reasons_spec -> reject_an_application_spec` and the feedback one too.

Comment (steventux) [reply spec/system/provider_interface/reject_an_application_spec.rb:8]: Perhaps I should remove all references to `redesign` now as it's not relevant?

**Review by cwrw — APPROVED** [review]
thanks! Should we hold off merging this until next week just incase?

**Comment by steventux** [comment]
@cwrw Yes sure, I think that's a good idea.

File: spec/system/provider_interface/reject_an_application_spec.rb (hunk 1 above)
Comment (cwrw) [reply spec/system/provider_interface/reject_an_application_spec.rb:8]: we can do that in the next cleanup PR might be easier, don't mind either way

**Pull request merged**